test/re/frida/MethodTest.java

```diff
@@ -39,6 +40,99 @@ public void replacementPropagatesExceptions() {
         thrown.expectMessage("Already dead");
         badger.die();
     }
+    
+    static private String failString = null;
+    static private void Fail( String msg )
+    {
+      failString = msg;
+    }
+    
+    @Test
+    public void TestNewInterface() {
+        loadScript("var X509TrustManager = Java.use('javax.net.ssl.X509TrustManager');" +
+                "try{" +
+                "  var tm = X509TrustManager.$new();" +
+                "}catch(e){" + 
+                "  var MethodTest = Java.use('re.frida.MethodTest');" +
+                "  MethodTest.Fail('couldnt create trustmanager: ' + e);" +
+                "}"
+                );
+        assertNull(failString);
+    }
+    
+    @Test
+    public void TestClassForName() {
+        loadScript("var c = Java.use('java.lang.Class');" +
+                "try{" +
+                "  var orig = c.forName.overload('java.lang.String');" +
+                "  c.forName.overload('java.lang.String').implementation = function(s){ orig(s); };" +
```

Comment:
For this test-suite we should avoid hooking things in the framework and only do that as a last resort. It's fine to start there though and refine the test so it's reproducing the same thing. The method in question here seems to be this one:

`static Class<?> forName(String className)`

And that made me think "uh oh, generics, we have some TODOs in that area", so maybe we can reproduce the same issue by using a locally defined method with the same kind of return type? I'd just throw it into the `Badger` class at the bottom.